### PR TITLE
Use pvc uid instead of pvc name in glusterfs endpoint name.

### PR DESCRIPTION
**Pull request by humblec**
GlusterFS: At present, endpoints and services are created
for glusterfs pvcs are in form of glusterfs-dynamic-<PVC name>
however this could cause issue if user deletes a PVC and immediately
creates a new one with the same name, PV controller will try to delete
the old PV and its endpoint and at the same the controller will try to create new PV
and the same endpoint. Depending on which event reaches the
controller first, it may create new PV, check that endpoints exists,
then delete the old PVC and delete endpoints already used by the new PV.

This patch changes the endpoint/service name to below format:

`glusterfs-dynamic-<PVC UUID>`.

By the uniqueness of UUID, above mentioned issue will no longer be present.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -822,11 +822,11 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 		return nil, 0, "", fmt.Errorf("failed to get cluster nodes for volume %s: %v", volume, err)
 	}
 
-	// The 'endpointname' is created in form of 'glusterfs-dynamic-<claimname>'.
+	// The 'endpointname' is created in form of 'glusterfs-dynamic-<PVC UUID>'.
 	// createEndpointService() checks for this 'endpoint' existence in PVC's namespace and
-	// If not found, it create an endpoint and service using the IPs we dynamically picked at time
+	// if not found, it create an endpoint and service using the IPs we dynamically picked at time
```
Comment (obnoxxx) [inline pkg/volume/glusterfs/glusterfs.go]: This is correct but actually not related to the topic of this patch.
So that comment fix might be a separate patch...

**Review by obnoxxx — CHANGES_REQUESTED** [review]
Thanks Humble, the change looks mostly good, just small comments inline.

And the commit message needs improvement:
Generally a good commit message should be in a format of:

> GlusterFS: short 1-line summary of what the patch does
> 
> optionally,
> more lines explaining
> the patch
> 
> Signed-off-by: ....

So important is: a blank line after initial line. And don't start with the explanation in the initial line but with what the patch does.

File: pkg/volume/glusterfs/glusterfs.go (hunk 1 above)
Comment (humblec) [reply pkg/volume/glusterfs/glusterfs.go]: Put it as  a seperate commit.

**Comment by humblec** [comment]
> So important is: a blank line after initial line. And don't start with the explanation in the initial line but with what the patch does.
>
It was there in the PR title, however forgot in the commit message. Addressed it latest patch set.

**Review by obnoxxx — APPROVED** [review]
LGTM, thanks!

**Comment by jsafrane** [comment]
lgtm-ish, please squash the patches please.

**Comment by obnoxxx** [comment]
@jsafrane writes:
> lgtm-ish, please squash the patches please.

Why is that? I explicitly asked @humblec to split the patches because the comment-fix is not part of the topic of the other patch. If you prefer to have that in a different PR then, I'd rather have a second PR than polluting commits with unrelated changes.

File: pkg/volume/glusterfs/glusterfs.go (hunk 1 above)
Comment (obnoxxx) [reply pkg/volume/glusterfs/glusterfs.go]: Thanks @humblec . I noticed there is another error in the line: `it create` --> `it creates`. Maybe we just skip this hunk from the PR and consider the separate PR #70306 for comment cleanup?

**Review by obnoxxx — CHANGES_REQUESTED** [review]
spotted one more typo in the comment, that you are fixing, and was remembered of my patches previously posted in #68343 to clean up the comments, now resurrected in #70306..
